src/timeMachine/components/view_options/GeoOptions.test.tsx

```diff
@@ -0,0 +1,188 @@
+          }
+          return setViewProperties(state, {
+            layers: mapTypeLayer,
+          })
+        case 'circleMap':
+          mapTypeLayer[0] = {
+            type: 'circleMap',
+            radiusField: 'magnitude',
+            radiusDimension: {label: 'Magnitude'},
+            colorDimension: {label: 'Duration'},
+            colorField: 'duration',
+            colors: [
+              {type: 'min', hex: '#ff00b3'},
+              {value: 50, hex: '#343aeb'},
+              {type: 'max', hex: '#343aeb'},
+            ],
+          }
+          return setViewProperties(state, {
+            layers: mapTypeLayer,
+          })
+        default:
+          return state
+      }
+*/
+const setup = (override = {}) => {
+  const geoStateView = createView('geo')
+
+  const props: GeoViewProperties = {
+    type: 'geo',
+    queries: [],
+    shape: 'chronograf-v2',
+    center: {
+      lat: 0,
+      lon: 0,
+    },
+    zoom: 6,
+    allowPanAndZoom: true,
+    detectCoordinateFields: true,
+    mapStyle: '',
+    note: '',
+    showNoteWhenEmpty: true,
+    colors: [],
+    layers: [
+      {
+        type: 'pointMap',
+        colorDimension: {label: 'Duration'},
+        colorField: 'duration',
+        colors: [
+          {type: 'min', hex: '#ff0000'},
+          {value: 50, hex: '#343aeb'},
+          {type: 'max', hex: '#343aeb'},
+        ],
+        isClustered: false,
+      },
+    ],
+    ...override,
+  }
+
+  return renderWithRedux(<GeoOptions {...props} />, () => {
+    const appState = {...mockAppState} as any
+    appState.timeMachines.timeMachines.de.view = geoStateView
+    return appState
+  })
+}
+
+describe('GeoOptions Geo Customization Options', () => {
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('changes the lat, lon, allowPanAndZoom, and zoom properties for the selected map type', () => {
+    const {getByTestId, store} = setup()
+    const latitude = getByTestId('geoLatitude')
+    const longitude = getByTestId('geoLongitude')
+    const zoom = getByTestId('geoZoom')
+    const allowPanAndZoom = getByTestId('geoAllowPanZoom')
+    const spyLatitudeAction = jest.spyOn(timeMachineActions, 'setLatitude')
+    const spyLongitudeAction = jest.spyOn(timeMachineActions, 'setLongitude')
+    const spyZoomAction = jest.spyOn(timeMachineActions, 'setZoomValue')
+
+    fireEvent.change(latitude, {target: {value: 40}})
+    fireEvent.change(longitude, {target: {value: 39}})
+    fireEvent.change(zoom, {target: {value: 12}})
+    fireEvent.click(allowPanAndZoom)
+    expect(spyLatitudeAction).toHaveBeenCalledWith(40)
+    expect(spyLongitudeAction).toHaveBeenCalledWith(39)
+    expect(spyZoomAction).toHaveBeenCalledWith(12)
+
+    const storeProps = store.getState().timeMachines.timeMachines.de.view
+      .properties as GeoViewProperties
+    expect(storeProps.allowPanAndZoom).toEqual(false)
+    expect(storeProps.center).toEqual({lat: 40, lon: 39})
+    expect(storeProps.zoom).toEqual(12)
+  })
```

Comment:
so i'm not sure if an integration test is the best course of action to test redux actions. the problem here is that we're deeply testing implementation, but not much behavior. we're basically saying "when this click happens, this function is called" which is a bit of antipattern. we want to assert something more along the lines of, "when a user clicks this button, the system's new state looks like this"

the difference is the case that tests behavior says "this is what i expect to happen" and the case that tests implementation says "this is what i expect you to do". tests of implementation have their place, but i don't think this is it. i think testing the reducer logic in a unit test will probably be more valuable. we can talk more about this in person though